static/app/__mocks__/api.tsx

```diff
@@ -41,7 +41,23 @@ type ResponseType = ApiNamespace.ResponseMeta & {
   url: string;
 };
 
-type MockResponse = [resp: ResponseType, mock: jest.Mock];
+type AsyncDelay = undefined | number;
+
+type AddMockResponseOptions = {
+  /**
+   * Whether to return mocked api responses directly, or with a setTimeout delay.
+   *
+   * Set to `null` to disable the async delay
+   * Set to a `number` which will be the amount of time (ms) for the delay
+   */
+  asyncDelay?: AsyncDelay;
+};
+
+type MockResponse = [
+  resp: ResponseType,
+  mock: jest.Mock,
+  options: AddMockResponseOptions
```

Comment:
Couldn't we fold these options into the `resp` payload? It already contains a the `match` key that refines how requests are matched.